src/main/scala/bigquery4s/BigQuery.scala

```diff
@@ -155,7 +155,17 @@ object BigQuery {
     scopes: Seq[String] = Seq(BigqueryScopes.BIGQUERY)
   ): BigQuery = {
 
-    val credential = using(new FileInputStream(serviceAccountJsonFilePath)) { in =>
+    BigQuery.fromServiceAccountJsonInputStream(new FileInputStream(serviceAccountJsonFilePath), transport, jsonFactory, scopes)
+  }
+
+  def fromServiceAccountJsonInputStream(
+    serviceAccountJsonInputStream: InputStream,
+    transport: HttpTransport = new NetHttpTransport,
+    jsonFactory: JsonFactory = new JacksonFactory,
+    scopes: Seq[String] = Seq(BigqueryScopes.BIGQUERY)
+  ): BigQuery = {
+
+    val credential = using(serviceAccountJsonInputStream) { in =>
```

Comment:
Ah, the passed resource is closed here. I don't think it's a good approach. Closing a resource near its creation is easy to maintain.